hs_core/hs_rdf.py

```diff
@@ -0,0 +1,358 @@
+from django.contrib.contenttypes.fields import GenericRelation
+from django.core.exceptions import ValidationError
+from rdflib import Graph, BNode
+from rdflib.collection import Collection
+from rdflib.namespace import Namespace, NamespaceManager, DC, DCTERMS, RDF, RDFS
+from rdflib.plugin import register
+from rdflib.plugins.serializers.rdfxml import XMLLANG, OWL_NS, XMLBASE
+from rdflib.plugins.serializers.xmlwriter import XMLWriter
+from rdflib.serializer import Serializer
+from rdflib.term import Literal, URIRef
+from rdflib.util import first
+
+HSTERMS = Namespace("http://hydroshare.org/terms/")
+RDFS1 = Namespace("http://www.w3.org/2000/01/rdf-schema#")
+
+NAMESPACE_MANAGER = NamespaceManager(Graph())
+NAMESPACE_MANAGER.bind('hsterms', HSTERMS, override=False)
+NAMESPACE_MANAGER.bind("rdfs1", RDFS1, override=False)
+NAMESPACE_MANAGER.bind('dc', DC, override=False)
+NAMESPACE_MANAGER.bind('dcterms', DCTERMS, override=False)
+
+
+class RDF_MetaData_Mixin(object):
+    """
+    A mixin for MetaData objects which store their metadata in generic relations.  If metadata outside of generic
+    relations need to be used, you may extend ingest_metadata and get_rdf_graph to include the other metadata elements
+    """
+
+    def rdf_subject(self):
+        raise NotImplementedError("RDF_Metadata_Mixin implementations must implement rdf_subject")
+
+    def rdf_metadata_subject(self):
+        raise NotImplementedError("RDF_Metadata_Mixin implementations must implement rdf_metadata_subject")
+
+    def rdf_type(self):
+        raise NotImplementedError("RDF_Metadata_Mixin implementations must implement rdf_type")
+
```

Comment:
It is probably useful to include corresponding RDF schema URLs such as https://www.w3.org/TR/rdf-schema/#ch_type in the comment for reference on mappings.

Reply:
Done